sequencing-server/src/routes/command-expansion.ts

```diff
@@ -219,7 +346,250 @@ commandExpansionRouter.post('/put-expansion-set', async (req, res, next) => {
+    context.sequenceTemplateDataLoader.load({ modelId }),
+    context.simulatedActivityInstanceBySeqIdBatchLoader.loadMany(seqIds.map(seqId => {
+      return { simulationDatasetId, seqId }
+    }))
+  ]);
+
+  //  2. Determine the language being used (SeqN vs. STOL)
+  //        Presently, we assume based on a database constraint, that all templates pulled for a given model/parcel combo have
+  //        the same language. While this constraint will remain true its exact enforcement and therefore implementation in SQL
+  //        and here may be subject to change.
+  if (sequenceTemplates.length === 0) {
+    throw new Error(
+      `POST /command-expansion/expand-all-sequence-templates: No sequence templates found for modelId=(${modelId}).`,
+    );
+  }
+
+  // Check that all languages are the same across all templates for this model
+  const languages = sequenceTemplates.map(template => template.language).reduce((previous, current, __, _) => {
+    if (previous.includes(current)) {
+      return previous;
+    }
+    else {
+      previous.push(current);
+      return previous;
+    }
+  }, [] as string[])
+
+  if (languages.length > 1) {
+    throw new Error(
+      `POST /command-expansion/expand-all-sequence-templates: Sequence templates found for modelId=(${modelId}) using different languages (${languages}).`
+    )
+  }
+
+  // Select the correct seqBuilder based on language
+  let seqBuilder: SeqBuilder<string, string>;
+  if (languages[0] === SequencingLanguage.STOL) {
+    seqBuilder = stolBuilder
+  } else if (languages[0] === SequencingLanguage.SEQN) {
+    seqBuilder = seqnBuilder
+  } else if (languages[0] === SequencingLanguage.TEXT) {
+    seqBuilder = concatBuilder
+  } else {
+    throw new Error(
+      `POST /command-expansion/expand-all-sequence-templates: Unsupported sequence language "${languages[0]}"`,
+    );
+  }
+
+  //  3. Pair seqId/SimulatedActivity lists; aggregate all simulated, filtered, activities
+  let seqIdToFilteredActivities: { [seqId: string]: { id: number, startOffset: Temporal.Duration }[] } = {};
+  let allFilteredActivities: { [id: number]: SimulatedActivity<Record<string, unknown>, Record<string, unknown>> } = [];
+
+  for (const entry of seqIds.entries()) {
+    let index = entry[0]
+    let seqId = entry[1]
+
+    // filteredActivities is a list of the SimulatedActivities for the current seqId
+    const filteredActivities = filteredSimulatedActivitiesBySeqId[index]
+    if (filteredActivities && !(filteredActivities instanceof Error)) {
+      // Extract just the id and start offset from each simulated activity
+      seqIdToFilteredActivities[seqId] = filteredActivities.map(act => {
+        return { id: act.id, startOffset: act.startOffset }
+      });
+
+      // Add this simulated activity to allFilteredActivities if it's not already there
+      // TODO: figure out whether we need to handle the case where a simulated activity is included with multiple
```

Comment:
This is a good point. The database does seem to permit this situation, but you're right that the frontend most likely doesn't. It's worth bringing up

Reply:
It does not and seems out of scope for now, as there aren't clear use cases. Is it okay to leave this TODO up, just so this functionality is clearly pointed out?

Reply:
Yes, let's leave the comment, but reword it to be explanatory rather than a TODO comment. Maybe something like this?

```
// NOTE: The database schema permits a simulated activity to be associated with multiple seq IDs, even though
// there is no way to create that multi-association using the UI. This code will honor the multi-association
```